integration-tests/jiva/replicadeletion/jiva_replica_deletion_test.go

```diff
@@ -0,0 +1,181 @@
+			artifacts.Artifact(jivaSCYaml),
+		)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		//Extracting PVC artifacts unstructured
+		pvcUnstructured, err := artifacts.GetArtifactUnstructured(
+			artifacts.Artifact(pvcYaml),
+		)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		//Apply jiva storageClass
+		cu := k8s.CreateOrUpdate(
+			k8s.GroupVersionResourceFromGVK(jivaSCUnstructured),
+			jivaSCUnstructured.GetNamespace(),
+		)
+
+		By("Deploying jiva-1r storageClass")
+		_, err = cu.Apply(jivaSCUnstructured)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		//Apply PVC using jiva storageClass in jiva-test namespace
+		cu = k8s.CreateOrUpdate(
+			k8s.GroupVersionResourceFromGVK(pvcUnstructured),
+			pvcUnstructured.GetNamespace(),
+		)
+
+		By("Deploying PVC using jiva-1r storageClass in jiva-test namespace")
+		_, err = cu.Apply(pvcUnstructured)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// pvcLabel represents the coressponding pvc
+		pvcLabel := "openebs.io/persistent-volume-claim=" + pvcUnstructured.GetName()
+		replicaLabel = defaultReplicaLabel + "," + pvcLabel
+		ctrlLabel = defaultCtrlLabel + "," + pvcLabel
+
+		// Verify creation of jiva ctrl pod
+		_ = checkPodUpandRunning(string(jivaTestNamespace), ctrlLabel, 1)
+
+		// Verify creation of jiva replica pod
+		_ = checkPodUpandRunning(string(jivaTestNamespace), replicaLabel, 1)
+	})
+
+	AfterEach(func() {
+		//Extracting PVC artifacts unstructured
+		pvcUnstructured, err := artifacts.GetArtifactUnstructured(
+			artifacts.Artifact(pvcYaml),
+		)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		//Extracting storageclass artifacts unstructured
+		jivaSCUnstructured, err := artifacts.GetArtifactUnstructured(
+			artifacts.Artifact(jivaSCYaml),
+		)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		//Delete the PVC using it's unstructured format
+		cu := k8s.DeleteResource(
+			k8s.GroupVersionResourceFromGVK(pvcUnstructured),
+			string(jivaTestNamespace),
+		)
+		By("Deleting PVC in jiva-test namespace")
+		err = cu.Delete(pvcUnstructured)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		//Delete jiva1-r storageClass
+		cu = k8s.DeleteResource(
+			k8s.GroupVersionResourceFromGVK(jivaSCUnstructured),
+			jivaSCUnstructured.GetNamespace(),
+		)
+		By("Deleting jiva-1r storageClass")
+		err = cu.Delete(jivaSCUnstructured)
+		Expect(err).ShouldNot(HaveOccurred())
+	})
+
+	Context("jiva replica deletion test", func() {
+		It("Deletion of jiva replica pod", func() {
+			By("Get the jiva replica pod details and perform test")
+			var nodeName, podName string
+			for i := 0; i < testTimes; i++ {
+				pods := checkPodUpandRunning(string(jivaTestNamespace), replicaLabel, 1)
+				// Deployed volume using single replica
+				podName = pods.Items[0].ObjectMeta.Name
+				if i == 0 {
+					// nodeName consist where the replica pod deployed
+					nodeName = pods.Items[0].Spec.NodeName
+				} else {
+					assert.Equal(GinkgoT(), pods.Items[0].Spec.NodeName, nodeName)
```

Comment:
Let us use gomega and avoid use of other assertion libraries

Reply:
Done